pravega/pravega
# Issue 5634: System Test Log Retention

**Change log description**  

* Adds a new step in the gradle system test deployment: `runFluentBitInstall`, which calls to a `fluentBitSetup.sh` script.
* On a call to the `startK8SystemTests` task, this script will setup fluent-bit and the PVC (using the default storageclass: `standard`) used to persist all logs generated by _all_ pods.

**Purpose of the change**  
Completes #5634.

**What the code does**  

* A `fluent-bit` DaemonSet is installed which will watch the logs of all pods and aggregate them into a single directory on the defined PVC. As configured, it avoids storing any extra information provided by fluent-bit or Kubernetes as apart of the logs. Meaning that fluent-bit stores _only_ the output generated by the application, and no extra metadata (as it does by default).

* An ad-hoc Kubernetes deployment is also deployed, which checks the current size of these logs at regular intervals, and rotates them once they exceed the defined threshold. When a rotation happens, it will will convert the compressed log's epoch suffix to a more human readable format (e.g ` 2021-02-16T00:17:10Z`).
 
The defaults are set to check at 10 second intervals and rotate when a log exceeds 10 MB.

**How to verify it**  

Run system tests, and verify that a pod can mount the created PVC and read the logs.
